site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -122,11 +128,13 @@ const CreateWorkspacePage: FC = () => {
     }
   });
 
+  const autoStartReady =
+    mode === "auto" && (!autofillEnabled || userParametersQuery.isSuccess);
```

Comment:
Defining this outside seemed like the more manageable thing to do, since this needs to evaluate twelve combinations:
- The value of mode (form, auto, duplicate)
- The value of autofillEnabled (true or false)
- The value of userParametersQuery.isSuccess (true or false)

Combining them into a single boolean makes sure the effect only has to worry about three situations:
1. Mounting render
2. Value flips from false to true
3. Value flips from true to false